ext/ext.go

```diff
@@ -0,0 +1,160 @@
+package ext
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"sort"
+	"strings"
+	"sync"
+)
+
+// TODO: Make an ExtensionRegistry?
+//
+//nolint:gochecknoglobals
+var (
+	mx         sync.RWMutex
+	extensions = make(map[ExtensionType]map[string]*Extension)
+)
+
+// ExtensionType is the type of all supported k6 extensions.
+type ExtensionType uint8
+
+// All supported k6 extension types.
+const (
+	JSExtension ExtensionType = iota + 1
+	OutputExtension
+)
+
+func (e ExtensionType) String() string {
+	var s string
+	switch e {
+	case JSExtension:
+		s = "js"
+	case OutputExtension:
+		s = "out"
+	}
+	return s
+}
+
+// Extension is a generic container for any k6 extension.
+type Extension struct {
+	Name, Path, Version string
+	Type                ExtensionType
+	Module              interface{}
```

Comment:
This comment might be caused by the fact that I'm only partially through the PR, but I must admit that looking at this field's definition and how it's used in `Register` as a user, I would be unclear as to what kind of value I should use to set the `Module` field.

Reply:
Sure, it is confusing. :disappointed: We initially called this "module" when JS extensions were introduced, as JS extensions would register a struct of their JS module. But then we changed the API for output extensions and had them register a constructor function that would return a module, for initialization purposes. In retrospect, I think this was the wrong API to introduce, but since I doubt we'll change this, every extension type is now free to choose what works best.

Since I can't think of a more generic name that would fit better, I'm inclined to leave it as `Module`.